### PR TITLE
CATTY-15 Implement WhenConditionScript

### PR DESCRIPTION
**Pull request by aeluna**
This Script executes all subsequent Bricks when a condition becomes true. In order to run it again later, the condition has to first become false once, and only once it becomes again true, the script will be run once more.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by KrennKristof — APPROVED** [review]

File: src/Catty/DataModel/Scripts/WhenConditionScript.swift [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import Foundation
+
+@objc(WhenConditionScript)
+class WhenConditionScript: Script, BrickFormulaProtocol {
+    @objc var condition: Formula!
```
Comment (m-herold) [inline src/Catty/DataModel/Scripts/WhenConditionScript.swift:27]: Please remove the force unwrapping.

**Review by m-herold — CHANGES_REQUESTED** [review]
Thanks! I found two problems while testing this PR:

(1) The Script should only execute once the condition was false once. Compare by running the following project on Catroid and Catty: https://share.catrob.at/app/project/48c24ee0-cb3d-11ea-a5c3-005056a36f47

(2) Open the script in the formula editor, type in a number and click on "Computer". The app crashes.

File: src/Catty/DataModel/Scripts/WhenConditionScript.swift (hunk 1 above)
Comment (aeluna) [reply src/Catty/DataModel/Scripts/WhenConditionScript.swift]: Thank you very much for your input! The problems should be fixed now.

**Review by m-herold — APPROVED** [review]

**Comment by m-herold** [comment]
Thank you @aeluna!

**Pull request merged**